[BUG] Serialization not supported for ListColumn

**Describe the bug**
Serialization is not currently supported for columns of `ListDtype`.  This prevents multi-gpu scaling with dask_cudf.

**Steps/Code to reproduce bug**
```python
import cudf

df = cudf.DataFrame({"a": [[1, 2, 3, None], [4, 5]]})
df.serialize()
```
**Trace**
```
python/cudf/cudf/core/dataframe.py:531: in serialize
    column_header, column_frames = column.serialize_columns(self._columns)
python/cudf/cudf/core/column/column.py:1994: in serialize_columns
    header_columns = [c.serialize() for c in columns]
python/cudf/cudf/core/column/column.py:1994: in <listcomp>
    header_columns = [c.serialize() for c in columns]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <cudf.core.column.lists.ListColumn object at 0x7fd58c493dd0>

    def serialize(self):
        header = {}
        frames = []
        header["type-serialized"] = pickle.dumps(type(self))
>       header["dtype"] = self.dtype.str
E       AttributeError: 'ListDtype' object has no attribute 'str'

python/cudf/cudf/core/column/column.py:1178: AttributeError
```

**Expected behavior**
Serialization should not result in an Error.

**Environment overview (please complete the following information)**
 - Environment location: Bare-metal
 - Method of cuDF install: conda

**Environment details**
<details><summary>Click here to see environment details</summary><pre>
     
     **git***
     commit 8dd9323c295a9ad8d7637a60db353fee13d7ba4a (HEAD, upstream/branch-0.17)
     Author: GALI PREM SAGAR <sagarprem75@gmail.com>
     Date:   Mon Nov 9 12:42:01 2020 -0600
     
     Fix issue when `numpy.str_` is given as input to string parameters in io APIs (#6701)
     
     Fixes: #6672
     
     This PR handlesnumpy.str_ inputs given to string parameters in cuio APIs by typecasting to str before passing to cython apis.
     **git submodules***
     
     ***OS Information***
     DGX_NAME="DGX Server"
     DGX_PRETTY_NAME="NVIDIA DGX Server"
     DGX_SWBUILD_DATE="2020-03-04"
     DGX_SWBUILD_VERSION="4.4.0"
     DGX_COMMIT_ID="ee09ebc"
     DGX_PLATFORM="DGX Server for DGX-1"
     DGX_SERIAL_NUMBER="QTFCOU8220024"
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=18.04
     DISTRIB_CODENAME=bionic
     DISTRIB_DESCRIPTION="Ubuntu 18.04.4 LTS"
     NAME="Ubuntu"
     VERSION="18.04.4 LTS (Bionic Beaver)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 18.04.4 LTS"
     VERSION_ID="18.04"
     HOME_URL="https://www.ubuntu.com/"
     SUPPORT_URL="https://help.ubuntu.com/"
     BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
     PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
     VERSION_CODENAME=bionic
     UBUNTU_CODENAME=bionic
     Linux dgx14 4.15.0-76-generic #86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
     
     ***GPU Information***
     Mon Nov  9 16:29:59 2020
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 440.64.00    Driver Version: 440.64.00    CUDA Version: 10.2     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |===============================+======================+======================|
     |   0  Tesla V100-SXM2...  On   | 00000000:06:00.0 Off |                    0 |
     | N/A   29C    P0    43W / 300W |      0MiB / 32510MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   1  Tesla V100-SXM2...  On   | 00000000:07:00.0 Off |                    0 |
     | N/A   31C    P0    40W / 300W |      0MiB / 32510MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   2  Tesla V100-SXM2...  On   | 00000000:0A:00.0 Off |                    0 |
     | N/A   30C    P0    42W / 300W |      0MiB / 32510MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   3  Tesla V100-SXM2...  On   | 00000000:0B:00.0 Off |                    0 |
     | N/A   28C    P0    41W / 300W |      0MiB / 32510MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   4  Tesla V100-SXM2...  On   | 00000000:85:00.0 Off |                    0 |
     | N/A   30C    P0    43W / 300W |      0MiB / 32510MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   5  Tesla V100-SXM2...  On   | 00000000:86:00.0 Off |                    0 |
     | N/A   30C    P0    41W / 300W |      0MiB / 32510MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   6  Tesla V100-SXM2...  On   | 00000000:89:00.0 Off |                    0 |
     | N/A   32C    P0    43W / 300W |      0MiB / 32510MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   7  Tesla V100-SXM2...  On   | 00000000:8A:00.0 Off |                    0 |
     | N/A   28C    P0    43W / 300W |      0MiB / 32510MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     
     +-----------------------------------------------------------------------------+
     | Processes:                                                       GPU Memory |
     |  GPU       PID   Type   Process name                             Usage      |
     |=============================================================================|
     |  No running processes found                                                 |
     +-----------------------------------------------------------------------------+
     
     ***CPU***
     Architecture:        x86_64
     CPU op-mode(s):      32-bit, 64-bit
     Byte Order:          Little Endian
     CPU(s):              80
     On-line CPU(s) list: 0-79
     Thread(s) per core:  2
     Core(s) per socket:  20
     Socket(s):           2
     NUMA node(s):        2
     Vendor ID:           GenuineIntel
     CPU family:          6
     Model:               79
     Model name:          Intel(R) Xeon(R) CPU E5-2698 v4 @ 2.20GHz
     Stepping:            1
     CPU MHz:             3295.414
     CPU max MHz:         3600.0000
     CPU min MHz:         1200.0000
     BogoMIPS:            4390.01
     Virtualization:      VT-x
     L1d cache:           32K
     L1i cache:           32K
     L2 cache:            256K
     L3 cache:            51200K
     NUMA node0 CPU(s):   0-19,40-59
     NUMA node1 CPU(s):   20-39,60-79
     Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti intel_ppin ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a rdseed adx smap intel_pt xsaveopt cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts md_clear flush_l1d
     
     ***CMake***
     /datasets/rzamora/miniconda3/envs/cudf_17/bin/cmake
     cmake version 3.18.2
     
     CMake suite maintained and supported by Kitware (kitware.com/cmake).
     
     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
     Copyright (C) 2017 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     
     
     ***nvcc***
     /usr/local/cuda/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2019 NVIDIA Corporation
     Built on Wed_Oct_23_19:24:38_PDT_2019
     Cuda compilation tools, release 10.2, V10.2.89
     
     ***Python***
     /datasets/rzamora/miniconda3/envs/cudf_17/bin/python
     Python 3.7.8
     
     ***Environment Variables***
     PATH                            : /home/nfs/rzamora/.vscode-server/bin/fcac248b077b55bae4ba5bab613fd6e9156c2f0c/bin:/home/nfs/rzamora/bin:/home/nfs/rzamora/.local/bin:/datasets/rzamora/miniconda3/envs/cudf_17/bin:/datasets/rzamora/miniconda3/condabin:/usr/local/cuda/bin:/opt/bin:/home/nfs/rzamora/.vscode-server/bin/fcac248b077b55bae4ba5bab613fd6e9156c2f0c/bin:/home/nfs/rzamora/bin:/home/nfs/rzamora/.local/bin:/usr/local/cuda/bin:/opt/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
     LD_LIBRARY_PATH                 :
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    : /datasets/rzamora/miniconda3/envs/cudf_17
     PYTHON_PATH                     :
     
     ***conda packages***
     /datasets/rzamora/miniconda3/condabin/conda
     # packages in environment at /datasets/rzamora/miniconda3/envs/cudf_17:
     #
     # Name                    Version                   Build  Channel
     _libgcc_mutex             0.1                 conda_forge    conda-forge
     _openmp_mutex             4.5                       1_gnu    conda-forge
     abseil-cpp                20200225.2           he1b5a44_2    conda-forge
     alabaster                 0.7.12                     py_0    conda-forge
     appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
     argon2-cffi               20.1.0           py37h8f50634_2    conda-forge
     arrow-cpp                 1.0.1           py37hd1d2865_14_cuda    conda-forge
     arrow-cpp-proc            2.0.0                      cuda    conda-forge
     async_generator           1.10                       py_0    conda-forge
     attrs                     20.2.0             pyh9f0ad1d_0    conda-forge
     aws-c-common              0.4.59               he1b5a44_0    conda-forge
     aws-c-event-stream        0.1.6                h84e28f3_5    conda-forge
     aws-checksums             0.1.9                he252421_2    conda-forge
     aws-sdk-cpp               1.8.63               h9b98462_0    conda-forge
     babel                     2.8.0                      py_0    conda-forge
     backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
     backports                 1.0                        py_2    conda-forge
     backports.functools_lru_cache 1.6.1                      py_0    conda-forge
     black                     19.10b0                    py_4    conda-forge
     bleach                    3.2.1              pyh9f0ad1d_0    conda-forge
     bokeh                     2.2.3            py37hc8dfbb8_0    conda-forge
     boost-cpp                 1.74.0               h9359b55_0    conda-forge
     brotli                    1.0.9                he1b5a44_3    conda-forge
     brotlipy                  0.7.0           py37hb5d75c8_1001    conda-forge
     bzip2                     1.0.8                h516909a_3    conda-forge
     c-ares                    1.16.1               h516909a_3    conda-forge
     ca-certificates           2020.6.20            hecda079_0    conda-forge
     certifi                   2020.6.20        py37he5f6b98_2    conda-forge
     cffi                      1.14.3           py37h00ebd2e_1    conda-forge
     cfgv                      3.2.0                      py_0    conda-forge
     chardet                   3.0.4           py37he5f6b98_1008    conda-forge
     clang                     8.0.1                hc9558a2_2    conda-forge
     clang-tools               8.0.1                hc9558a2_2    conda-forge
     clangxx                   8.0.1                         2    conda-forge
     click                     7.1.2              pyh9f0ad1d_0    conda-forge
     cloudpickle               1.6.0                      py_0    conda-forge
     cmake                     3.18.2               h3f3948e_0    conda-forge
     cmake_setuptools          0.1.3                      py_0    rapidsai
     commonmark                0.9.1                      py_0    conda-forge
     cryptography              3.1.1            py37hff6837a_1    conda-forge
     cudatoolkit               10.2.89              h6bb024c_0    nvidia
     cudf                      0.17.0a0+192.g8dd9323c29.dirty           dev_0    <develop>
     cudnn                     7.6.5                cuda10.2_0
     cupy                      8.0.0            py37hd9eba22_2    conda-forge
     cython                    0.29.21          py37hb892b2f_1    conda-forge
     cytoolz                   0.11.0           py37h8f50634_1    conda-forge
     dask                      2.30.0+44.g7beeebbb           dev_0    <develop>
     dask-cuda                 0.17.0a0+40.ge9aee6c           dev_0    <develop>
     dask-cudf                 0.17.0a0+192.g8dd9323c29.dirty           dev_0    <develop>
     decorator                 4.4.2                      py_0    conda-forge
     defusedxml                0.6.0                      py_0    conda-forge
     distlib                   0.3.1              pyh9f0ad1d_0    conda-forge
     distributed               2.31.0.dev0+25.g49044c3c           dev_0    <develop>
     dlpack                    0.3                  he1b5a44_1    conda-forge
     docutils                  0.16             py37he5f6b98_2    conda-forge
     double-conversion         3.1.5                he1b5a44_2    conda-forge
     editdistance              0.5.3            py37h3340039_2    conda-forge
     entrypoints               0.3             py37hc8dfbb8_1002    conda-forge
     expat                     2.2.9                he1b5a44_2    conda-forge
     fastavro                  1.0.0.post1      py37h8f50634_1    conda-forge
     fastrlock                 0.5              py37h3340039_1    conda-forge
     filelock                  3.0.12             pyh9f0ad1d_0    conda-forge
     flake8                    3.8.3                      py_1    conda-forge
     flatbuffers               1.12.0               he1b5a44_0    conda-forge
     freetype                  2.10.4               he06d7ca_0    conda-forge
     fsspec                    0.8.4                      py_0    conda-forge
     future                    0.18.2           py37hc8dfbb8_2    conda-forge
     gflags                    2.2.2             he1b5a44_1004    conda-forge
     glog                      0.4.0                h49b9bf7_3    conda-forge
     gmp                       6.2.0                he1b5a44_3    conda-forge
     grpc-cpp                  1.32.0               h7997a97_1    conda-forge
     heapdict                  1.0.1                      py_0    conda-forge
     hypothesis                5.38.1                     py_0    conda-forge
     icu                       67.1                 he1b5a44_0    conda-forge
     identify                  1.5.6              pyh9f0ad1d_0    conda-forge
     idna                      2.10               pyh9f0ad1d_0    conda-forge
     imagesize                 1.2.0                      py_0    conda-forge
     importlib-metadata        2.0.0                      py_1    conda-forge
     importlib_metadata        2.0.0                         1    conda-forge
     iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
     ipykernel                 5.3.4            py37hc6149b9_1    conda-forge
     ipython                   7.18.1           py37hc6149b9_1    conda-forge
     ipython_genutils          0.2.0                      py_1    conda-forge
     isort                     5.0.7            py37hc8dfbb8_0    conda-forge
     jedi                      0.17.2           py37hc8dfbb8_1    conda-forge
     jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
     jpeg                      9d                   h516909a_0    conda-forge
     json5                     0.9.5              pyh9f0ad1d_0    conda-forge
     jsonschema                3.2.0                      py_2    conda-forge
     jupyter_client            6.1.7                      py_0    conda-forge
     jupyter_core              4.6.3            py37hc8dfbb8_2    conda-forge
     jupyterlab                2.2.9                      py_0    conda-forge
     jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
     jupyterlab_server         1.2.0                      py_0    conda-forge
     krb5                      1.17.1               hfafb76e_3    conda-forge
     lcms2                     2.11                 hbd6801e_0    conda-forge
     ld_impl_linux-64          2.35                 h769bd43_9    conda-forge
     libblas                   3.9.0                2_openblas    conda-forge
     libcblas                  3.9.0                2_openblas    conda-forge
     libcurl                   7.71.1               hcdd3856_8    conda-forge
     libedit                   3.1.20191231         he28a2e2_2    conda-forge
     libev                     4.33                 h516909a_1    conda-forge
     libevent                  2.1.10               hcdb4288_3    conda-forge
     libffi                    3.2.1             he1b5a44_1007    conda-forge
     libgcc-ng                 9.3.0               h5dbcf3e_17    conda-forge
     libgfortran-ng            9.3.0               he4bcb1c_17    conda-forge
     libgfortran5              9.3.0               he4bcb1c_17    conda-forge
     libgomp                   9.3.0               h5dbcf3e_17    conda-forge
     liblapack                 3.9.0                2_openblas    conda-forge
     libllvm10                 10.0.1               he513fc3_3    conda-forge
     libllvm8                  8.0.1                hc9558a2_0    conda-forge
     libnghttp2                1.41.0               h8cfc5f6_2    conda-forge
     libopenblas               0.3.12          pthreads_h4812303_1    conda-forge
     libpng                    1.6.37               hed695b0_2    conda-forge
     libprotobuf               3.13.0.1             h8b12597_0    conda-forge
     librmm                    0.17.0a201022   cuda10.2_g210c863_36    rapidsai-nightly
     libsodium                 1.0.18               h516909a_1    conda-forge
     libssh2                   1.9.0                hab1572f_5    conda-forge
     libstdcxx-ng              9.3.0               h2ae2ef3_17    conda-forge
     libthrift                 0.13.0               h5aa387f_6    conda-forge
     libtiff                   4.1.0                hc7e4089_6    conda-forge
     libutf8proc               2.5.0                h516909a_2    conda-forge
     libuv                     1.40.0               hd18ef5c_0    conda-forge
     libwebp-base              1.1.0                h516909a_3    conda-forge
     llvmlite                  0.34.0           py37h5202443_2    conda-forge
     locket                    0.2.0                      py_2    conda-forge
     lz4-c                     1.9.2                he1b5a44_3    conda-forge
     markdown                  3.3.3              pyh9f0ad1d_0    conda-forge
     markupsafe                1.1.1            py37hb5d75c8_2    conda-forge
     mccabe                    0.6.1                      py_1    conda-forge
     mimesis                   4.0.0              pyh9f0ad1d_0    conda-forge
     mistune                   0.8.4           py37h8f50634_1002    conda-forge
     more-itertools            8.5.0                      py_0    conda-forge
     msgpack-python            1.0.0            py37h99015e2_2    conda-forge
     nbclient                  0.5.1                      py_0    conda-forge
     nbconvert                 6.0.7            py37hc8dfbb8_1    conda-forge
     nbformat                  5.0.8                      py_0    conda-forge
     nbsphinx                  0.7.1              pyh9f0ad1d_0    conda-forge
     nccl                      2.7.8.1              hc6a2c23_1    conda-forge
     ncurses                   6.2                  he1b5a44_2    conda-forge
     nest-asyncio              1.4.1                      py_0    conda-forge
     nodeenv                   1.5.0              pyh9f0ad1d_0    conda-forge
     notebook                  6.1.4            py37hc8dfbb8_1    conda-forge
     numba                     0.51.2           py37h9fdb41a_0    conda-forge
     numpy                     1.19.2           py37h7008fea_1    conda-forge
     numpydoc                  1.1.0                      py_1    conda-forge
     nvtx                      0.2.1            py37h8f50634_2    conda-forge
     olefile                   0.46               pyh9f0ad1d_1    conda-forge
     openssl                   1.1.1h               h516909a_0    conda-forge
     orc                       1.6.5                hd3605a7_0    conda-forge
     packaging                 20.4               pyh9f0ad1d_0    conda-forge
     pandas                    1.1.3            py37hb33c840_2    conda-forge
     pandoc                    1.19.2                        0    conda-forge
     pandocfilters             1.4.2                      py_1    conda-forge
     parquet-cpp               1.5.1                         2    conda-forge
     parso                     0.7.1              pyh9f0ad1d_0    conda-forge
     partd                     1.1.0                      py_0    conda-forge
     pathspec                  0.8.0              pyh9f0ad1d_0    conda-forge
     pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
     pickleshare               0.7.5                   py_1003    conda-forge
     pillow                    8.0.1            py37h718be6c_0    conda-forge
     pip                       20.2.4                     py_0    conda-forge
     pluggy                    0.13.1           py37he5f6b98_3    conda-forge
     pre-commit                2.7.1            py37hc8dfbb8_1    conda-forge
     pre_commit                2.7.1                         1    conda-forge
     prometheus_client         0.8.0              pyh9f0ad1d_0    conda-forge
     prompt-toolkit            3.0.8                      py_0    conda-forge
     protobuf                  3.13.0.1         py37h3340039_1    conda-forge
     psutil                    5.7.3            py37hb5d75c8_0    conda-forge
     ptyprocess                0.6.0                   py_1001    conda-forge
     py                        1.9.0              pyh9f0ad1d_0    conda-forge
     py-cpuinfo                7.0.0              pyh9f0ad1d_0    conda-forge
     pyarrow                   1.0.1           py37h5b20ac3_14_cuda    conda-forge
     pycodestyle               2.6.0              pyh9f0ad1d_0    conda-forge
     pycparser                 2.20               pyh9f0ad1d_2    conda-forge
     pyflakes                  2.2.0              pyh9f0ad1d_0    conda-forge
     pygments                  2.7.2                      py_0    conda-forge
     pynvml                    8.0.4                    pypi_0    pypi
     pyopenssl                 19.1.0                     py_1    conda-forge
     pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
     pyrsistent                0.17.3           py37h8f50634_1    conda-forge
     pysocks                   1.7.1            py37he5f6b98_2    conda-forge
     pytest                    6.1.1            py37he5f6b98_1    conda-forge
     pytest-benchmark          3.2.3              pyh9f0ad1d_0    conda-forge
     python                    3.7.8           h6f2ec95_1_cpython    conda-forge
     python-dateutil           2.8.1                      py_0    conda-forge
     python_abi                3.7                     1_cp37m    conda-forge
     pytz                      2020.1             pyh9f0ad1d_0    conda-forge
     pyyaml                    5.3.1            py37hb5d75c8_1    conda-forge
     pyzmq                     19.0.2           py37hac76be4_2    conda-forge
     rapidjson                 1.1.0             he1b5a44_1002    conda-forge
     re2                       2020.10.01           he1b5a44_0    conda-forge
     readline                  8.0                  he28a2e2_2    conda-forge
     recommonmark              0.6.0                      py_0    conda-forge
     regex                     2020.10.23       py37h8f50634_0    conda-forge
     requests                  2.24.0             pyh9f0ad1d_0    conda-forge
     rhash                     1.3.6             h516909a_1001    conda-forge
     rmm                       0.17.0a201022   cuda_10.2_py37_g210c863_36    rapidsai-nightly
     send2trash                1.5.0                      py_0    conda-forge
     setuptools                49.6.0           py37he5f6b98_2    conda-forge
     six                       1.15.0             pyh9f0ad1d_0    conda-forge
     snappy                    1.1.8                he1b5a44_3    conda-forge
     snowballstemmer           2.0.0                      py_0    conda-forge
     sortedcontainers          2.2.2              pyh9f0ad1d_0    conda-forge
     spdlog                    1.7.0                hc9558a2_2    conda-forge
     sphinx                    3.2.1                      py_0    conda-forge
     sphinx-copybutton         0.3.0              pyh9f0ad1d_0    conda-forge
     sphinx-markdown-tables    0.0.14             pyh9f0ad1d_1    conda-forge
     sphinx_rtd_theme          0.5.0              pyh9f0ad1d_0    conda-forge
     sphinxcontrib-applehelp   1.0.2                      py_0    conda-forge
     sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
     sphinxcontrib-htmlhelp    1.0.3                      py_0    conda-forge
     sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
     sphinxcontrib-qthelp      1.0.3                      py_0    conda-forge
     sphinxcontrib-serializinghtml 1.1.4                      py_0    conda-forge
     sphinxcontrib-websupport  1.2.4              pyh9f0ad1d_0    conda-forge
     sqlite                    3.33.0               h4cf870e_1    conda-forge
     streamz                   0.6.0                    pypi_0    pypi
     tblib                     1.6.0                      py_0    conda-forge
     terminado                 0.9.1            py37hc8dfbb8_1    conda-forge
     testpath                  0.4.4                      py_0    conda-forge
     tk                        8.6.10               hed695b0_1    conda-forge
     toml                      0.10.1             pyh9f0ad1d_0    conda-forge
     toolz                     0.11.1                     py_0    conda-forge
     tornado                   6.0.4            py37h8f50634_2    conda-forge
     traitlets                 5.0.5                      py_0    conda-forge
     typed-ast                 1.4.1            py37h516909a_0    conda-forge
     typing_extensions         3.7.4.3                    py_0    conda-forge
     urllib3                   1.25.11                    py_0    conda-forge
     virtualenv                20.1.0           py37hc8dfbb8_0    conda-forge
     wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
     webencodings              0.5.1                      py_1    conda-forge
     wheel                     0.35.1             pyh9f0ad1d_0    conda-forge
     xz                        5.2.5                h516909a_1    conda-forge
     yaml                      0.2.5                h516909a_0    conda-forge
     zeromq                    4.3.3                he1b5a44_2    conda-forge
     zict                      2.0.0                      py_0    conda-forge
     zipp                      3.4.0                      py_0    conda-forge
     zlib                      1.2.11            h516909a_1010    conda-forge
     zstd                      1.4.5                h6597ccf_2    conda-forge
     
</pre></details>

**Additional context**
This bug prevents multi-GPU handling of multi-hot categoricals in NVTabular (see [nvtabular#419](https://github.com/NVIDIA/NVTabular/issues/419))

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -94,6 +94,7 @@
 - PR #6693 Fix issue related to `na_values` input in `read_csv`
 - PR #6701 Fix issue when `numpy.str_` is given as input to string parameters in io APIs
 - PR #6704 Fix leak warnings in JNI unit tests
+- PR #6721 Add missing serialization methods for ListColumn
 
 
 # cuDF 0.16.0 (21 Oct 2020)

--- a/python/cudf/cudf/core/column/lists.py
+++ b/python/cudf/cudf/core/column/lists.py
@@ -1,9 +1,11 @@
 # Copyright (c) 2020, NVIDIA CORPORATION.
 
 import pyarrow as pa
+import pickle
 
 import cudf
-from cudf.core.column import ColumnBase
+from cudf.core.buffer import Buffer
+from cudf.core.column import column, ColumnBase
 from cudf.core.column.methods import ColumnMethodsMixin
 from cudf.utils.dtypes import is_list_dtype
 
@@ -107,6 +109,58 @@ class ListColumn(ColumnBase):
             )
         else:
             super().set_base_data(value)
+
+    def serialize(self):
+        header = {}
+        frames = []
+        header["type-serialized"] = pickle.dumps(type(self))
+        header["dtype"] = pickle.dumps(self.dtype)
+        header["null_count"] = self.null_count
+        header["size"] = pickle.dumps(self.size)
+
+        frames = []
+        sub_headers = []
+
+        for item in self.children:
+            sheader, sframes = item.serialize()
+            sub_headers.append(sheader)
+            frames.extend(sframes)
+
+        if self.null_count > 0:
+            frames.append(self.mask)
+
+        header["subheaders"] = sub_headers
+        header["frame_count"] = len(frames)
+
+        return header, frames
+
+    @classmethod
+    def deserialize(cls, header, frames):
+
+        # Get null mask
+        if header["null_count"] > 0:
+            mask = Buffer(frames[-1])
+        else:
+            mask = None
+
+        # Deserialize child columns
+        children = []
+        f = 0
+        for h in header["subheaders"]:
+            fcount = h["frame_count"]
+            child_frames = frames[f:f+fcount]
+            column_type = pickle.loads(h["type-serialized"])
+            children.append(column_type.deserialize(h, child_frames))
+            f += fcount
+
+        # Materialize list column
+        return column.build_column(
+            data=None,
+            dtype=pickle.loads(header["dtype"]),
+            mask=mask,
+            children=tuple(children),
+            size=pickle.loads(header["size"]),
+        )
 
 
 class ListMethods(ColumnMethodsMixin):

--- a/python/cudf/cudf/core/column/lists.py
+++ b/python/cudf/cudf/core/column/lists.py
@@ -117,7 +117,7 @@ class ListColumn(ColumnBase):
         header["type-serialized"] = pickle.dumps(type(self))
         header["dtype"] = pickle.dumps(self.dtype)
         header["null_count"] = self.null_count
-        header["size"] = pickle.dumps(self.size)
+        header["size"] = self.size
 
         frames = []
         sub_headers = []
@@ -160,7 +160,7 @@ class ListColumn(ColumnBase):
             dtype=pickle.loads(header["dtype"]),
             mask=mask,
             children=tuple(children),
-            size=pickle.loads(header["size"]),
+            size=header["size"],
         )
 
 

--- a/python/cudf/cudf/core/column/lists.py
+++ b/python/cudf/cudf/core/column/lists.py
@@ -1,11 +1,12 @@
 # Copyright (c) 2020, NVIDIA CORPORATION.
 
-import pyarrow as pa
 import pickle
+
+import pyarrow as pa
 
 import cudf
 from cudf.core.buffer import Buffer
-from cudf.core.column import column, ColumnBase
+from cudf.core.column import ColumnBase, column
 from cudf.core.column.methods import ColumnMethodsMixin
 from cudf.utils.dtypes import is_list_dtype
 
@@ -148,7 +149,7 @@ class ListColumn(ColumnBase):
         f = 0
         for h in header["subheaders"]:
             fcount = h["frame_count"]
-            child_frames = frames[f:f+fcount]
+            child_frames = frames[f : f + fcount]
             column_type = pickle.loads(h["type-serialized"])
             children.append(column_type.deserialize(h, child_frames))
             f += fcount

--- a/python/cudf/cudf/core/column/string.py
+++ b/python/cudf/cudf/core/column/string.py
@@ -4780,7 +4780,7 @@ class StringColumn(column.ColumnBase):
     def serialize(self):
         header = {"null_count": self.null_count}
         header["type-serialized"] = pickle.dumps(type(self))
-        header["size"] = pickle.dumps(self.size)
+        header["size"] = self.size
 
         frames = []
         sub_headers = []
@@ -4799,7 +4799,7 @@ class StringColumn(column.ColumnBase):
 
     @classmethod
     def deserialize(cls, header, frames):
-        size = pickle.loads(header["size"])
+        size = header["size"]
         # Deserialize the mask, value, and offset frames
         buffers = [Buffer(each_frame) for each_frame in frames]
 

--- a/python/cudf/cudf/tests/test_serialize.py
+++ b/python/cudf/cudf/tests/test_serialize.py
@@ -265,3 +265,23 @@ def test_serialize_string_check_buffer_sizes():
     header, frames = df.serialize()
     got = sum(b.nbytes for b in frames)
     assert expect == got
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {"a": [[]]},
+        {"a": [[1, 2, None, 4]]},
+        {"a": [["cat", None, "dog"]]},
+        {
+            "a": [[1, 2, 3, None], [4, None, 5]],
+            "b": [None, ["fish", "bird"]],
+            "c": [[], []],
+        },
+        {"a": [[1, 2, 3, None], [4, None, 5], None, [6, 7]]},
+    ],
+)
+def test_serialize_list_columns(data):
+    df = cudf.DataFrame(data)
+    recreated = df.__class__.deserialize(*df.serialize())
+    assert_eq(recreated, df)

